src/Std/Time/Format.lean

```diff
@@ -0,0 +1,287 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+prelude
+import Std.Time.Notation
+import Std.Time.Format.Basic
+import Std.Time.Internal.Bounded
+
+namespace Std
+namespace Time
+namespace Formats
+
+set_option linter.all true
+
+/--
+The ISO8601 format, which is always 24 or 27 characters long, used for representing date and time in
+a standardized format. The format follows the pattern `YYYY-MM-DD'T'hh:mm:ss'Z'`.
+-/
+def ISO8601 : Format .any := date-spec% "YYYY-MM-DD'T'hh:mm:ss.sssZ"
+
+/--
+The AmericanDate format, which follows the pattern `MM/DD/YYYY`.
+-/
+def AmericanDate : Format .any := date-spec% "MM-DD-YYYY"
+
+/--
+The EuropeanDate format, which follows the pattern `DD/MM/YYYY`.
+-/
+def EuropeanDate : Format .any := date-spec% "DD-MM-YYYY"
+
+/--
+The Time12Hour format, which follows the pattern `HH:mm:ss aa` for representing time
+in a 12-hour clock format with an AM/PM marker.
```

Comment:
Should specify here that the AM/PM marker is lowercase as you currently have it. I would argue that uppercase would be better though.